### PR TITLE
Test coverage

### PR DESCRIPTION
**Pull request by dblodgett-usgs**
It's been bugging me ... more to come. If any watchers want to take a look at review, I'd appreciate it.

### CODE REVIEW

**Comment by dblodgett-usgs** [comment]
With log in, my coverage report sorted by missed lines is: 
image
image

File: R/AAA.R [hunk 1]
```diff
@@ -28,15 +28,6 @@ default_timeout <- function(){
...
 
-# helper fxn to pull out items by name, using either lapply or vapply
-pluck <- function(x, name, type) {
-	if (missing(type)) {
-		lapply(x, "[[", name)
-	} else {
-		vapply(x, "[[", name, FUN.VALUE = type)
-	}
-}
-
```
Comment (dblodgett-usgs) [inline R/AAA.R:39]: This function isn't used.

File: R/item_get.R [hunk 2]
```diff
@@ -28,3 +28,14 @@
...
+get_item <- function(id, ..., session=current_session()) {
+	res <- sbtools_GET(url = paste0(pkg.env$url_item, id), ..., 
+										 query = list(type = 'json'), session = session)
+	
+	if(is(res, "list")) {
+		if(res$status == 404) return(NULL)
+	}
+	
+	return(as.sbitem(content(res)))
+}
```
Comment (dblodgett-usgs) [inline R/item_get.R:41]: Moved this here because it just made sense to put it with item_get.

File: R/item_list_files.R [hunk 3]
```diff
@@ -41,27 +41,29 @@ item_list_files = function(sb_id, recursive = FALSE, ..., session=current_sessio
...
-					if (is(w, "data.frame")) w$id else w
-				})
-				if (!is(item[[1]], "sbitem")) item <- unlist(item, FALSE)
-				item <- lapply(item, item_list_children, session = session)
-				children <- lapply(item, function(z) lapply(z$id, as.sbitem))
-				out[[i]] <- children
-				kids <- any(comp(sapply(unlist(children, FALSE), "[[", "hasChildren")))
+				
+				message(paste("Checked child item", i - 1, "found", length(next_children), "more children."))
+
```
Comment (dblodgett-usgs) [inline R/item_list_files.R:62]: The code above wasn't working. I rewrite it and it is now tested.

File: R/item_upload_files.R [hunk 4]
```diff
@@ -0,0 +1,81 @@
...
+	for(i in 1:length(files)){
+		if(!file.exists(files[i])){
+			stop('This file does not exist or cannot be accessed: ', files[i])
+		}
+		
+		body[[paste0('file', i)]] = upload_file(files[i])
+	}
+	names(body) = rep('file', length(body))
+	return(body)
+}
```
Comment (dblodgett-usgs) [inline R/item_upload_files.R:81]: None of the code here changed, I just consolidated it into a single file so my brain didn't hurt so bad.

File: R/query_sb_spatial.R [hunk 5]
```diff
@@ -26,13 +26,8 @@
...
 	if(!missing(bbox)){
-		if(requireNamespace("sp")){
-			#project first into longlat projection so bounding box is in degrees
-			bb = sp::spTransform(bbox, sp::CRS("+proj=longlat +datum=WGS84"))@bbox
-			bb_wkt = make_wkt(bb)
-		}else{
-			stop('sp package required to use bbox parameter. Please: install.packages("sp")')
-		}
+		message("bbox must be from an sp object in WGS84 Lon/Lat")
+		bb_wkt = make_wkt(bbox@bbox)
```
Comment (dblodgett-usgs) [inline R/query_sb_spatial.R:30]: The above was requiring rgdal for the spTransform. I just raise a message now.

File: R/sb_item.R [hunk 6]
```diff
@@ -66,22 +68,3 @@ is.sbitem <- function(x) {
...
-get_item <- function(id, ..., session=current_session()) {
-	res <- sbtools_GET(url = paste0(pkg.env$url_item, id), ..., 
-									 query = list(type = 'json'), session = session)
-	
-	if(is(res, "list")) {
-		if(res$status == 404) return(NULL)
-	}
-	
-	return(as.sbitem(content(res)))
-}
```
Comment (dblodgett-usgs) [inline R/sb_item.R:87]: Pluck wasn't used. and get_item was moved.

File: tests/testthat/test-REST.R [hunk 7]
```diff
@@ -38,3 +38,27 @@ test_that("generic get w/ and w/o auth", {
...
+	
+	expect_warning(sbtools_GET("https://www.sciencebase.go/catalog/item/5c4f4a04e4b0708288f78e07", session = NULL))
+	
+	put_test <- sbtools_PUT("https://www.sciencebase.gov/catalog/item/5c4f4a04e4b0708288f78e07", "test", session = session)
+	
+	expect_equal(put_test$status_code, 405)
+	
+	delete_test <- sbtools_DELETE("https://www.sciencebase.gov/catalog/item/5c4f4a04e4b0708288f78e07", session = session)
+
+	expect_equal(delete_test$status_code, 405)
```
Comment (wdwatkins) [inline tests/testthat/test-REST.R:61]: So with the 405 expected, these are just testing that the package has at least tried the put/delete?

**Review by wdwatkins — COMMENTED** [review]
Seems good to me, although I not very knowledgable about the package.

File: tests/testthat/test-REST.R (hunk 7 above)
Comment (dblodgett-usgs) [reply tests/testthat/test-REST.R:61]: Yeah -- I'm just trying to get some lines covered so we have something. More testing coming later that will only run in local dev when logged in that will do more here.

**Pull request merged**